rust/src/utils.rs

```diff
@@ -66,22 +66,55 @@ where
     Ok(parsed)
 }
 
-fn download_url_to_tmpfile(url: &str) -> Result<fs::File> {
-    let mut tmpf = tempfile::tempfile()?;
-    {
-        let mut output = io::BufWriter::new(&mut tmpf);
-        let mut handle = Easy::new();
-        handle.follow_location(true)?;
-        handle.fail_on_error(true)?;
-        handle.url(url)?;
+pub(crate) fn download_url_to_tmpfile(url: &str, progress: bool) -> Result<fs::File> {
+    // Unwrap safety: multiple case returns the same number of values
+    Ok(download_urls_to_tmpfiles(vec![url], progress)?
+        .into_iter()
+        .next()
+        .expect("file"))
```

Comment:
Minor: this looks like a simple `Ok(tmpfiles[0])`, I think?

Reply:
I get a borrowck error that I can't move a `!Copy` type out of the `Vec`.

Reply:
You are right, nevermind.